esphome/components/lvgl/lvgl_esphome.h

```diff
@@ -319,6 +347,40 @@ class LvglComponent : public PollingComponent {
 
   void add_updater(Updater *updater) { this->updaters_.push_back(updater); }
 
+#if USE_TOUCHSCREEN
+  void add_touchscreen(Touchscreen *touchscreen) {
+    LVTouchListener tsl;
+    lv_indev_drv_t drv;
+    lv_indev_drv_init(&drv);
+    drv.type = LV_INDEV_TYPE_POINTER;
+    drv.read_cb = [tsl](lv_indev_drv_t *drv, lv_indev_data_t *data) { tsl.touch_cb(data); };
+    lv_indev_drv_register(&drv);
+    touchscreen->register_listener(&tsl);
+  }
+#endif
+#if USE_ROTARY
+  void add_rotary(Sensor *sensor, Sensor *binary_sensor, std::string group) {
+    static bool pressed_{false};
+    static uint32_t count_{0}, last_{0};
+    lv_indev_drv_t drv;
+    lv_indev_drv_init(&drv);
+    drv.type = LV_INDEV_TYPE_ENCODER;
+    drv.read_cb = [pressed_, count_, &last_](lv_indev_drv_t *drv, lv_indev_data_t *data) {
+      data->state = pressed_ ? LV_INDEV_STATE_PRESSED : LV_INDEV_STATE_RELEASED;
+      data->enc_diff = count_ - last_;
+      last_ = count_;
+    };
+    lv_indev_drv_register(&drv);
```

Comment:
You can't call this here. It has to be called after the display is created. See what I did with touchscreens, where this is added to the init lambda.